plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHiveFailureRecovery.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import io.trino.Session;
+import io.trino.operator.RetryPolicy;
+import io.trino.testing.AbstractTestFailureRecovery;
+import io.trino.testing.QueryRunner;
+import io.trino.tpch.TpchTable;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public abstract class AbstractTestHiveFailureRecovery
```

Comment:
Similar question here, why do we have both `AbstractTestHiveFailureRecovery` and `AbstractTestFailureRecovery`? I don't fully understand why we define some of tests in `AbstractTestFailureRecovery` and again override in `AbstractTestHiveFailureRecovery`. Do you plan to have other classes extending  `AbstractTestFailureRecovery` in the future?

Reply:
AbstractTestFailureRecovery is not Hive specific. It may be used as a subclass for test class for Iceberg or other connectors.